src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc

```diff
@@ -562,15 +554,15 @@ GrpcLb::Picker::PickResult GrpcLb::Picker::Pick(PickState* pick,
       abort();
     }
     grpc_mdelem lb_token = {reinterpret_cast<uintptr_t>(arg->value.pointer.p)};
-    AddLbTokenToInitialMetadata(GRPC_MDELEM_REF(lb_token),
-                                &pick->lb_token_mdelem_storage,
-                                pick->initial_metadata);
-    // Pass on client stats via context. Passes ownership of the reference.
-    if (client_stats_ != nullptr) {
-      pick->subchannel_call_context[GRPC_GRPCLB_CLIENT_STATS].value =
-          client_stats_->Ref().release();
-      pick->subchannel_call_context[GRPC_GRPCLB_CLIENT_STATS].destroy =
-          DestroyClientStats;
+    GPR_ASSERT(!GRPC_MDISNULL(lb_token));
+    GPR_ASSERT(grpc_metadata_batch_add_tail(pick->initial_metadata,
+                                            &pick->lb_token_mdelem_storage,
+                                            GRPC_MDELEM_REF(lb_token))
+                    == GRPC_ERROR_NONE);
+    GrpcLbClientStats* client_stats = static_cast<GrpcLbClientStats*>(
+        grpc_mdelem_get_user_data(lb_token, GrpcLbClientStats::Destroy));
+    if (client_stats != nullptr) {
+      client_stats->AddCallStarted();
```

Comment:
Previously, we were calling this in the constructor of the call_data. Are there any implications to calling it here instead?